### ISSUE TITLE
Let users copy/paste the master key

### ISSUE THREAD

**Issue by Tbaut** [cyph/cyph#71]
Hi,
Just a quick heads-up. While I find cyph so far very user-friendly, congrats for that, I found it annoying not to be able to copy-paste the master key. Even worse, because I had to inspect the site's DOM content to end up copy pasting it, I copied it with white space before and after, not sure why that, but 

Suggestion: let us copy/paste, add a copy button, and use padding-left/-right rather than white space for design :)

Although you mention [here](https://github.com/cyph/cyph/blob/62cbd9ebb1dd1c90633d8d0a1d09264dedf4e690/shared/js/cyph/components/account-register/account-register.component.html#L1341-L1357) to put it into a file or on a cloud storage, not even mentioning it should be encrypted, and not in plain text, you mention "don't store it into a password manager". Why so? I don't think this is right. A password manager has its flaws, but it is orders of magnitude more secure and adapted to store a master key than "a file" on a computer.

**Comment by eganist** [comment]
@Tbaut good afternoon!

The text on the link you referenced reads:

> Don't print it out, save it to cloud storage, or save it to a file. Don't even store it in an encrypted password manager. Write it down on paper and keep it well hidden.

in English, this first sentence is interpreted as a list to which "don't" is equally applied:

> Don't
> 
> * print it out
> 
> * save it to cloud storage
>
> * or save it to a file.

Hopefully this clarifies your initial reading.

As for password managers, it depends on your threat model. The presumption Cyph is making is that users may have an easier time protecting their master key on paper than using technology where side channel attacks and phishing are commonplace. But in the end, if you decide that a password manager works with your threat model, you should do it.

**Comment by buu700** [comment]
Yeah, we've gotten this request a lot recently, and I think we'll go ahead and do it after making the user click through a warning.

The concern we have re: password managers is specifically online password managers, as they tend to have a number of issues and you would be effectively reducing the security of your Cyph account to the security of that password managers. However, a local password manager is basically fine, although paper is still the recommended approach.

**Comment by Tbaut** [comment]
> in English, this first sentence is interpreted as a list to which "don't" is equally applied

That makes much more sense, although I def. know this, I misread it. That's a nit but if I read it incorrectly, I guess others will. Repeating the "don't" wouldn't hurt to make things clear, after all, we're talking about the most sensitive piece of information your users will have to take care of.

I understand your point about password managers, they're not all equal and I certainly didn't open this issue to start an endless conversation about it :) I did it just to give my feedback. I learnt from the many crypto-wallet I build, that you can do whatever you want to force your users to write a seed phrase on a piece of paper, if they don't want to, they won't, and will either leave your app, or work around it.

> I think we'll go ahead and do it after making the user click through a warning

This sounds good to me :+1:
